ui/tests/integration/image-file-test.js

```diff
@@ -0,0 +1,111 @@
+import { find, settled } from '@ember/test-helpers';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+import RSVP from 'rsvp';
+import { formatBytes } from 'nomad-ui/helpers/format-bytes';
+
+module('Integration | Component | image file', function(hooks) {
+  setupRenderingTest(hooks);
+
+  const commonTemplate = hbs`
+    {{image-file src=src alt=alt size=size}}
+  `;
+
+  const commonProperties = {
+    // HACK!
+    // Without mocking the entire image element, this component will make a
+    // real request for a real image. It's unfortuante to couple this test to the
+    // favicon, but it's an unlikely asset to be removed, and it's better than
+    // attempting to mock this.
+    src: '/ui/favicon.png',
+    alt: 'This is the alt text',
+    size: 123456,
+  };
+
```

Comment:
Would using a data-uri or a Blob here? Loading an svg using a data-uri would be the simplest I think. I'm not sure what you need to do exactly here so I might be going down the wrong path - but I think a datauri would work for what you need..maybe..?

Reply:
Ah cool, thank you both for investigating. I assumed poorly that a data-uri wouldn't trigger the onload event, but in hindsight it makes sense that the API is consistently abstract from the data scheme.